docs/syntax_and_semantics/enum.md

```diff
@@ -117,37 +119,22 @@ Class variables are allowed, but instance variables are not.
 
 ## Usage
 
-Enums are a type-safe alternative to [Symbol](https://crystal-lang.org/api/Symbol.html). For example, an API's method can specify a [type restriction](type_restrictions.md) using an enum type:
+When a method parameter has an enum [type restriction](type_restrictions.md), it accepts either an enum constant or a Symbol. The Symbol will be automatically cast to an enum constant, raising a compile-time error if casting fails.
 
 ```crystal
 def paint(color : Color)
-  case color
-  when Color::Red
-    # ...
-  else
-    # Unusual, but still can happen
-    raise "unknown color: #{color}"
-  end
+  puts "Painting using the color #{color}"
 end
 
 paint Color::Red
-```
-
-The above could also be implemented with a Symbol:
 
-```crystal
-def paint(color : Symbol)
-  case color
-  when :red
-    # ...
-  else
-    raise "unknown color: #{color}"
-  end
-end
+paint :red # automatically casts to `Color::Red`
 
-paint :red
+# compile-time error:
+#   expected argument #1 to 'paint' to match a member of enum Color
+#
+#   Options are: :red, :green and :blue
+paint :yellow
```

Comment:
Errors are usually formatted like this:
```suggestion
paint :yellow # Error: expected argument #1 to 'paint' to match a member of enum Color
```

I think it's sufficient to show the main error message and skip the extra help here.